### PR TITLE
Add catalan stemmer.

### PR DESCRIPTION
**Pull request by miceno**
Using the resources available at Snowball web site: https://snowballstem.org/algorithms/catalan/stemmer.html

I will appreciate you will consider integrating it into the main branch.

Let me know if there is anything I should change to adapt it to this repo best practices.

### CODE REVIEW

**Comment by wamania** [comment]
Thanks for this new stemmer !
It looks like good, so many suffix oO
I don't understand the purpose of CsvFileVerboseIterator
It's not important, but even if i let (for now) some "array()", you should use [ ] for better readability (not really require for merge).
The stem file for tests come from the official website ?
Plz update the readme (you can put your name on the catalan stemmer line if you want)

File: src/Stemmer/Catalan.php [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    /**
+     * Step 0: Attached pronoun
+     *
+     * Search for the longest among the following suffixes
+     * and delete it in R1.
+     */
+
+    private function step0()
+    {
+        if (($position = $this->search(static::$attached_pronoun)) != false) {
```
Comment (wamania) [inline src/Stemmer/Catalan.php]: !== 
everywhere
search use strrpos, so it doesn't mind, but good practice for this case

**Review by wamania — CHANGES_REQUESTED** [review]

File: src/Stemmer/Catalan.php (hunk 1 above)
Comment (miceno) [reply src/Stemmer/Catalan.php]: Good! I will do.

**Comment by miceno** [comment]
Thank you!!!

> Thanks for this new stemmer !
> It looks like good, so many suffix oO

Yes, latin languages are so.

> I don't understand the purpose of CsvFileVerboseIterator

I run phpunittest and when it finds an error, it shows up as "error at \#323", showing the 0-based line number inside the testing file. It was really annoying, since I had to open the file and go to the line to show the word and the like. So I read a bit about PHPUnit loaders and `Iterator`, so they allow setting the key of the iterator to a more meaningful value that is used by the loader code to show up in test description.

> It's not important, but even if i let (for now) some "array()", you should use [ ] for better readability (not really require for merge).

Sure, no problem.

> The stem file for tests come from the official website ?

Yes, they are from Snowball site.

> Plz update the readme (you can put your name on the catalan stemmer line if you want)

Sure, I will do.

**Comment by miceno** [comment]
> > Plz update the readme (you can put your name on the catalan stemmer line if you want)

How should I include it? Something like this?

- Catalan (by Orestes Sanchez <orestes@estotienearreglo.es>).

**Comment by miceno** [comment]
Concerning testing, I saw PHPUnit version for PHPStemmer 2.x is still on version 4.8... see `composer.json` file.

Have you consider upgrading to a higher PHPUnit version? Is there any incompatibility?

**Comment by wamania** [comment]
> Concerning testing, I saw PHPUnit version for PHPStemmer 2.x is still on version 4.8... see `composer.json` file.
> 
> Have you consider upgrading to a higher PHPUnit version? Is there any incompatibility?

i still with old namespace "PHPUnit_Framework_TestCase"
but i should upgrade of course, i'll think about it

**Comment by miceno** [comment]
Catalan (by Orestes Sanchez <orestes@estotienearreglo.es>).


> > Concerning testing, I saw PHPUnit version for PHPStemmer 2.x is still on version 4.8... see `composer.json` file.
> > Have you consider upgrading to a higher PHPUnit version? Is there any incompatibility?
> 
> i still with old namespace "PHPUnit_Framework_TestCase"
> but i should upgrade of course, i'll think about it

I will give it a try, in another PR, I was asking just to avoid trying something useless :)

**Comment by miceno** [comment]
Changes applied :)

**Comment by wamania** [comment]
Capture du 2020-05-25 23-06-16

no error with standard csv iterator

PS : sorry, i just understand, it's helpful just for the dev phase, to see errors

**Comment by wamania** [comment]
ok, i keep it
should be in a "dev doc", i will add that.
I merge, thank you !

**Comment by wamania** [comment]
> I will give it a try, in another PR, I was asking just to avoid trying something useless :)

it will be welcome

**Pull request merged**